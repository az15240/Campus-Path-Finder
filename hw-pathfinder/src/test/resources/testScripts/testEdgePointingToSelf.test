# Test a special edge that connects from and to the same node.
# Graph A consists of three nodes, each having such a special edge.
# Graph B consists of three nodes, two of them have a special edge, and one edge connecting
# these two nodes, and one edge connecting one of these nodes and the other node.

CreateGraph A
AddNode A n1
AddEdge A n1 n1 1
ListChildren A n1
AddNode A n2
AddEdge A n2 n2 2
ListChildren A n2
AddNode A n3
AddEdge A n3 n3 3
ListChildren A n3
ListNodes A

CreateGraph B
AddNode B n1
AddEdge B n1 n1 10
ListChildren B n1
AddNode B n2
AddEdge B n2 n2 20
ListChildren B n2
AddEdge B n1 n2 12
ListChildren B n1
AddNode B n3
AddEdge B n1 n3 13
ListChildren B n1
ListNodes B

FindPath A n1 n1
FindPath B n1 n3