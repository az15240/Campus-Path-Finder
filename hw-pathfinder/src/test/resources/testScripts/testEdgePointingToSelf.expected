# Test a special edge that connects from and to the same node.
# Graph A consists of three nodes, each having such a special edge.
# Graph B consists of three nodes, two of them have a special edge, and one edge connecting
# these two nodes, and one edge connecting one of these nodes and the other node.

created graph A
added node n1 to A
added edge 1.000 from n1 to n1 in A
the children of n1 in A are: n1(1.000)
added node n2 to A
added edge 2.000 from n2 to n2 in A
the children of n2 in A are: n2(2.000)
added node n3 to A
added edge 3.000 from n3 to n3 in A
the children of n3 in A are: n3(3.000)
A contains: n1 n2 n3

created graph B
added node n1 to B
added edge 10.000 from n1 to n1 in B
the children of n1 in B are: n1(10.000)
added node n2 to B
added edge 20.000 from n2 to n2 in B
the children of n2 in B are: n2(20.000)
added edge 12.000 from n1 to n2 in B
the children of n1 in B are: n1(10.000) n2(12.000)
added node n3 to B
added edge 13.000 from n1 to n3 in B
the children of n1 in B are: n1(10.000) n2(12.000) n3(13.000)
B contains: n1 n2 n3

path from n1 to n1:
total cost: 0.000
path from n1 to n3:
n1 to n3 with weight 13.000
total cost: 13.000